tests/test_build.py

```diff
@@ -540,30 +540,41 @@ def test_edit_page_url(sphinx_build_factory, html_context, edit_url):
     assert edit_link[0].attrs["href"] == edit_url, f"edit link didn't match {edit_link}"
 
 
-def test_new_google_analytics_id(sphinx_build_factory):
-    confoverrides = {"html_theme_options.google_analytics_id": "G-XXXXX"}
-    sphinx_build = sphinx_build_factory("base", confoverrides=confoverrides)
-    sphinx_build.build()
-    index_html = sphinx_build.html_tree("index.html")
-
-    # Search all the scripts and make sure one of them has the Google tag in there
-    tags_found = False
-    for script in index_html.select("script"):
-        if script.string and "gtag" in script.string and "G-XXXXX" in script.string:
-            tags_found = True
-    assert tags_found is True
-
-
-def test_old_google_analytics_id(sphinx_build_factory):
-    confoverrides = {"html_theme_options.google_analytics_id": "UA-XXXXX"}
+@pytest.mark.parametrize(
+    "provider,tags",
+    [
+        # deprecated
+        # new_google_analytics_id
+        ({"html_theme_options.google_analytics_id": "G-XXXXX"}, ["gtag", "G-XXXXX"]),
+        # old_google_analytics_id
+        ({"html_theme_options.google_analytics_id": "UA-XXXXX"}, ["ga", "UA-XXXXX"]),
+        # google analytics
+        (
+            {"html_theme_options.analytics": {"google_analytics_id": "G-XXXXX"}},
+            ["gtag", "G-XXXXX"],
+        ),
+        # plausible
+        (
+            {
+                "html_theme_options.analytics": {
+                    "plausible_analytics_domain": "toto",
+                    "plausible_analytics_url": "http://.../script.js",
+                }
+            },
+            ["data-domain", "toto"],
+        ),
+    ],
+)
+def test_analytics(sphinx_build_factory, provider, tags):
```

Comment:
this parameterize is beautiful, thanks for updating it in this way

Reply:
Thanks 😄